### ISSUE TITLE
Component for in-app web view

### ISSUE THREAD

**Issue by bessgreen** [department-of-veterans-affairs/va.gov-team#14164]
### Acceptance Criteria

1. Build out a generic component for in-app web display
2. Top nav options consist of:
    - **Done**: close out the in-app site so the user is viewing the mobile app page
    - **Title**: dynamic. For most in-app web views, this will be "va.gov", but there will be others including "veteranscrisisline.net"
    - **Refresh**
3. Bottom nav options consist of:
    - **L/R arrows**
    - **Open in browser**

### Designs
https://adhoc.invisionapp.com/share/YUYRT9BAPSB#/screens/431636866
![Screen Shot 2020-10-08 at 9.17.45 AM.png](https://images.zenhubusercontent.com/5f64e41291069a617acc1b47/7487c76d-0b4e-44a6-be75-ab5a4e26330b)

### Dev checklist

- [x] UX/UI
- [ ] Accessibility
- [ ] Add tests

**Comment by jjulian** [comment]
For in-app webviews such as facility locator, I think we'll want it to look like all other screens. Standard nav bar, title, back button, etc. The web view should be 100% width. Deferring details to @bridgethapner and @lexicalninja

**Comment by bridgethapner** [comment]
@jjulian Appreciate your thoughts here! Looking into it a bit more, I'm inclined to agree. The big question that I'm curious about what other mobile designers think about is: have they seen users be confused by a feature being dressed up too much to look like part of the app when it's on an outside website? 

I did a little reading after SU and it seems like the biggest plus of going to the out of app browser for Facility Locator is enabling users to bookmark it and return to it outside of the app, as Brandon mentioned. The biggest downside seems to be a user needing to navigate out of their browser back to the app to return. 

Given these trade-offs, I'm inclined to keep the homepage items (Facility Locator, Coronavirus FAQs, Veterans Crisis Line resource button) as in-app web views to keep people in the app. 

Looking forward to talking more later.

**Comment by bridgethapner** [comment]
@bessgreen I updated wireframes for what I'd ideally like the web view to look like to answer the open Qs here. Here's [Facility Locator](https://adhoc.invisionapp.com/share/YUYRT9BAPSB#/screens/432493111) and here's [Coronavirus FAQs](https://adhoc.invisionapp.com/share/YUYRT9BAPSB#/screens/432493468). 

**Top nav options:**
- L: Done
- Center: Title (va.gov -- I think we should be clear about where the user is)
- R: Text resizing
- R: Refresh

**Bottom browser nav options:**
- L/R arrows
- Bookmark
- Open in browser

Please lmk if there's anything else you need to unblock this! 

cc @jjulian @jmarchi

**Comment by bessgreen** [comment]
Thanks, @bridgethapner! I touched base with @jmarchi and here's where we landed with each of the nav options:

**Top nav options**
- Done: this will close out the in-app site so the user is viewing the mobile app page
- Title: we can hardcode this to va.gov
- Text resizing: we'll need to do some more investigation here. Any solution we have will have to work with the va.gov site, so it'll require some coordination with their team
- Refresh: we can do this

**Bottom browser nav options**
- L/R arrows: we can do this
- Bookmark: since this isn't opening in a browser, there won't be anything to bookmark the page to. If a user wants to bookmark, they can open the link in their browser and bookmark it there. 
     - Follow-up Q: the icon you used looks similar to the native share/actions menu - as an alternate option, we could do something with this. Just let us know!
- Open in browser: we can do this

So to summarize, everything but Text Resizing (more investigation needed) and Bookmark should be good to go.  

cc @jjulian

**Comment by bridgethapner** [comment]
@bessgreen and John - Thank you! 

- Good to know about text resizing, I see that as a nice to have, we're okay either way for MVP. If this is high LOE let's backlog it. 
- I meant "Share" instead of "Bookmark", good catch! 
- Re hardcoding VA.gov: I think this is okay for MVP. 

Also related to web views, as an FYI: since this ticket is focused on the in-app webview, I created #14200 to make a list of all the links out of the app + define what type of view we're using for each. Plan to tackle the next sprint.

**Comment by bessgreen** [comment]
Update on nav options for MVP (per conversation with @bridgethapner):
 - Nav options will include: Done, Title, Refresh, L/R arrows, Open in browser
 - Text resizing and Actions/Share are out-of-scope for this story

**Comment by bridgethapner** [comment]
Sharing this thread @jjulian started about how we could hide the VA.gov header/footer in our VA.gov web views using some post-load javascript - https://dsva.slack.com/archives/C018V2JCWRJ/p1601575572124600. 

Would this javascript work be considered part of this front-end ticket or merit its own?

**Comment by bessgreen** [comment]
@bridgethapner great, thanks for linking that thread! I rolled the js work into the AC for this story. If for some reason it doesn't work and we need to track down another solution, we'll raise that with you and @jjulian to create a new task so it doesn't block the basic webview story.

**Comment by bridgethapner** [comment]
svgs attached

[https://app.zenhub.com/files/133843125/34f4ae3a-7b59-41d1-9dd2-774d24198483/download](https://app.zenhub.com/files/133843125/34f4ae3a-7b59-41d1-9dd2-774d24198483/download)

**Comment by bridgethapner** [comment]
svgs of lock and refresh icon @jmarchi [https://app.zenhub.com/files/133843125/46b6b1ba-1b0a-4744-8210-b5646e2d3559/download](https://app.zenhub.com/files/133843125/46b6b1ba-1b0a-4744-8210-b5646e2d3559/download)

**Comment by bessgreen** [comment]
Creating a separate [ticket](https://app.zenhub.com/workspaces/va-mobile-app-5f205eac3155a2000f422ecb/issues/department-of-veterans-affairs/va.gov-team/14906) to hide header/footer

**Comment by digitalken** [comment]
Jumped the gun on logging this bug against #14163.  I'll review this bug below when this has been code reviewed.
Issue(s):
https://app.zenhub.com/workspaces/va-mobile-app-5f205eac3155a2000f422ecb/issues/department-of-veterans-affairs/va.gov-team/14997

**Comment by digitalken** [comment]
Testing:

1.Verify Deque or similar tool finds no issues on navigation controls.

**Comment by digitalken** [comment]
Linked issue:

https://app.zenhub.com/workspaces/va-mobile-app-5f205eac3155a2000f422ecb/issues/department-of-veterans-affairs/va.gov-team/15053

**Comment by digitalken** [comment]
Added to TestRail.

**Comment by digitalken** [comment]
Pass 11/11/2020 - iOS and Android. Sending to Design QA.
